keps/sig-release/20190316-rebase-images-to-distroless.md

```diff
@@ -0,0 +1,167 @@
+- Another reason of using debian is from the CVE concerns. Those images are originally rebased from busybox to debian for better CVE feeds and management (See [PR/70245](https://github.com/kubernetes/kubernetes/pull/70245)).
+- A third type of images uses debian for certain external dependencies.
+
+#### Type 3 Alpine Based Images
+
+The reasons for images based on alpine are similar to the ones on debian. Debian is more widely used due to previous “Establish base image policy for system containers” effort (see [issue/40248](https://github.com/kubernetes/kubernetes/issues/40248)).
+
+### Distroless and Previous Work
+
+"Distroless images contain only your application and its runtime dependencies. They do not contain package managers, shells or any other programs you would expect to find in a standard Linux distribution.”  (See [distroless/README](https://github.com/GoogleContainerTools/distroless))
+Distroless supports the dependencies where “FROM scratch” misses and more light-weighted than debian or alpine. Meanwhile, distroless “improves the signal to noise of scanners (e.g. CVE) and reduces the burden of establishing provenance to just what you need.”(from [distroless/README](https://github.com/GoogleContainerTools/distroless))
+
+Using Distroless/static as a common image base is originally proposed as an exploration area in [the base image policy for system containers](https://github.com/kubernetes/kubernetes/issues/40248). Tim(tallclair@) has driven the effort on defining and establishing the base image policy (main changes):
+
+* Add Alpine iptable as base image for kube-proxy. Previously kube-proxy is based on debian iptable image.  (This direction is scrapped, see [issue/39696](https://github.com/kubernetes/kubernetes/issues/39696) for details)
+* Rebase busybox images to debian-base.
+* Rebase certain alpine images to debian-base
+
+The distroless/static solution is filed separately in [issue/70249](https://github.com/kubernetes/kubernetes/issues/70249). This kep, as a more up-to-date version, is slightly different than the original issue.
+
+## Proposal
+
+The approaches to rebase different containers can vary significantly due to the function of the containers, the cloud-providers’ release workflows, and legacy reasons (repo migration plans, retirement plans, etc). This section will discuss 4 main types of image rebasing strategies, and this should cover the majority of the kubernetes containers.
+
+1. The images are built via bazel. In such case, we will update the bazel BUILD rule to switch to the base image to distroless/static. This method applies for the core containers like kube-apiserver, kube-controller-manager, cloud-controller-manager, kube-scheduler. (See detailed solution in *Core Master Images*)
```

Comment:
I'm pretty sure every image that is built with bazel also has a non-bazel (typically via Make) counterpart. IIRC, it's because bazel doesn't support cross architecture builds, but that might be out of date. In either both the build rule & corresponding makefiles need to be updated.

Reply:
Ah, I see this is addressed below. Might be worth mentioning it here though.

Reply:
hmmm? What do you mean mentioning it here? I have a "See detailed solution ...". 

Changed that to a link.